### PR TITLE
Mark & recall scrolls (cross-level teleport home)

### PR DESCRIPTION
**Pull request by c0ze**
## Summary
Plan 15u — the teleport-home pair from 0.40's scroll list (`teleport.c`):

- **Mark scroll** (`cast_mark`): pins the current level + position ("Destination marked.").
- **Recall scroll** (`cast_recall`): snaps the reader back to the pin — **across levels**, re-entering through the Dungeon's persisted cache exactly like stairs Travel, so the marked level's state (dead monsters, dropped loot) is intact on arrival. "You find yourself back in <level>."
- Deliberate divergences, documented in the plan: an **unmarked recall fizzles** (the C would land at its zero-initialized location); a mark **occupied by a creature nudges** the player one ring outward (the C stacks creatures on one tile).
- Names per `treasure.c:1192/1199`: "mark scroll" / "recall scroll" — no "scroll of" prefix, same as blink.

## Test plan
- Same-level mark→wander→recall returns exactly to the pin; cross-level via a real two-level dungeon (mark on A, stairs to B, recall lands on A at the pin).
- Unmarked recall returns false / "nothing happens"; an occupied pin lands the player exactly one tile off.
- Behavior messages match the C; goldens for both prefix-less names.
- gofmt + go vet + full `go test ./...` green (10/10 packages).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added mark and recall scrolls; players can mark a location and teleport back, including across dungeon levels.
  * Recall will nudge you if the marked tile is occupied and fails gracefully if unmarked or fully blocked.

* **Documentation**
  * Added implementation plan for mark-and-recall mechanics.

* **Tests**
  * Added unit and integration coverage for mark/recall behaviors and edge cases.

### CODE REVIEW

File: go/internal/game/teleport.go [hunk 1]
```diff
@@ -42,3 +42,50 @@ func (g *Game) ForgetMap() {
...
+			for dx := -1; dx <= 1 && !nudged; dx++ {
+				p := Pos{X: dst.X + dx, Y: dst.Y + dy}
+				if p != dst && g.Level.Passable(p) && g.Level.CreatureAt(p) == nil {
+					dst, nudged = p, true
+				}
+			}
+		}
+		if !nudged {
+			return false // the mark is buried in bodies; nowhere to appear
+		}
```
[the remark this answers is not in the document]
Comment (c0ze) [reply go/internal/game/teleport.go]: Good catch — applied. Recall now resolves the landing spot via freeSpotNear before committing, and a buried mark rolls the level switch back (re-entering a cached level can't fail), so the 'nothing happens' report is honest. Regression test added (TestBuriedMarkRollsBackLevelSwitch).

File: go/internal/behaviors/behaviors_test.go [hunk 2]
```diff
@@ -372,3 +372,39 @@ func TestAmnesiaWipesTheMap(t *testing.T) {
...
+	if len(msgs) == 0 || !strings.Contains(msgs[0], "You find yourself back") {
+		t.Errorf("expected the C recall message, got %v", msgs)
+	}
+}
+
+func TestUnmarkedRecallBehaviorFizzles(t *testing.T) {
+	recall := Registry()["recall"]
+	floor := &content.TileDef{ID: "floor", Glyph: ".", Color: content.ColorNormal, Passable: true, Transparent: true}
+	g := &game.Game{Level: game.NewLevel(8, 3, floor), Player: game.Pos{X: 1, Y: 1}}
+	msgs := recall(g, &game.Item{Def: &content.ItemDef{Name: "recall scroll"}})
```
[the remark this answers is not in the document]
Comment (c0ze) [reply go/internal/behaviors/behaviors_test.go]: Applied — the lookup now uses the ok-guard pattern like the rest of the file.
